src/main/java/com/bootcamp/demo/restapi/BookingsController.java

```diff
@@ -0,0 +1,178 @@
+            return new ResponseEntity<>(bookingsByUserId, HttpStatus.OK);
+        } catch (ExecutionException | InterruptedException | IllegalArgumentException e) {
+            LOGGER.info("Unfortunately,an error happened while trying to retrive a booking based on the user id.");
+            return new ResponseEntity<>(bookingsByUserId, HttpStatus.BAD_REQUEST);
+        }
+    }
+
+    @GetMapping("/getBooking/{id}")
+    @ResponseBody
+    public ResponseEntity<Object> getBooking(@PathVariable(value = "id") final String id) {
+        Booking booking = new Booking();
+        try {
+            LOGGER.info("GET BOOKING BY ID - API endpoint invoked");
+
+            booking = bookingService.getBookingByID(id);
+            return new ResponseEntity<>(booking, HttpStatus.OK);
+        } catch (ExecutionException | InterruptedException | IllegalArgumentException e) {
+            LOGGER.info("Unfortunately,an error happened while trying to get a booking.");
+
+            return new ResponseEntity<>(booking, HttpStatus.BAD_REQUEST);
+        }
+    }
+
+    @GetMapping("/showAllBookings")
+    @ResponseBody
+    public ResponseEntity<LinkedHashSet<Booking>> getAllBookings() {
+        LinkedHashSet<Booking> bookings = new LinkedHashSet<>();
+        try {
+            bookings = bookingService.getAllBookings();
+            return new ResponseEntity<>(bookings, HttpStatus.OK);
+        } catch (ExecutionException | InterruptedException e) {
+            LOGGER.info("Unfortunately,an error happened while trying to get all the bookings.");
+
+            return new ResponseEntity<>(bookings, HttpStatus.BAD_REQUEST);
+        }
+    }
+
+    @PostMapping("/bookScooter")
+    @ResponseBody
+    public void bookScooter(final Double lat, final Double longitude) throws ExecutionException, InterruptedException {
+        bookAScooter(lat, longitude);
+
+    }
+
+    @DeleteMapping("/delete")
+    public ResponseEntity<Object> deleteBooking(@RequestParam final String id) {
+        try {
+            LOGGER.info("DELETE BOOKING - API endpoint invoked");
+            bookingService.deleteBooking(id);
+            return new ResponseEntity<>(HttpStatus.OK);
+        } catch (ExecutionException | InterruptedException | IllegalArgumentException e) {
+            LOGGER.info("Unfortunately,an error happened when trying to delete a booking.");
+            return new ResponseEntity<>(HttpStatus.BAD_REQUEST);
+        }
+    }
+
+    @PutMapping("/update")
+    public ResponseEntity<Object> updateBooking(@RequestParam final String id, @RequestParam("end")
+            String endDate, PaymentStatus status) {
+        try {
+            LOGGER.info("UPDATE BOOKING - API endpoint invoked");
+
+            Booking booking = bookingService.getBookingByID(id);
+            Scooter scooter = scooterService.findScooterById(booking.getSerialNumber());
+
+            if (status == PaymentStatus.SUCCESS) {

```

Comment:
Should the response be different based on whether the update was done or not? To be able to identify this also in swagger/client.